### ISSUE TITLE
[A11y] Dialog under blocking mode - screen reader cannot read the title or subtitle of the dialog 

### ISSUE THREAD

**Issue by rulyu** [OfficeDev/office-ui-fabric-react#10391]
### Environment Information

- **Package version(s)**: 7.15.0
- **Browser and OS versions**: chrome with NVDA screen reader

### Describe the issue:

When the `isBlocking: true` is set for `modalProps`, screen reader does not read the title and the subtitle of the screen.
#### Actual behavior:

When the `isBlocking: true` is set for `modalProps`, screen reader does not read the title and the subtitle of the dialog.
#### Expected behavior:

Screen reade should read the title and the subtitle of the dialog regardless of the `isBlocking` prop.
### If applicable, please provide a codepen repro:
https://codepen.io/ruoxi/pen/aboYbjx

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: this issue is negatively impacting our accessibility test grade. We would like to see a fix asap.

Products/sites affected: https://csi.ai.dynamics.com

**Comment by natalieethell** [comment]
Hey @rulyu I'm able to repro what you're seeing using NVDA on Chrome. I looked into it a bit to see if it was a Fabric Dialog issue or an NVDA issue, and it seems to be NVDA-specific. 

Here's what I found - one of the main differences between the Dialog with `isBlocking = true` and the Dialog with `isBlocking = false` is the role. A non-blocking Dialog has `role = dialog` and a blocking Dialog has `role = alertdialog`.

If you look at the [`alertdialog` example](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/alertdialog.html) on the W3C Standards site, you'll see that Narrator on Edge reads out the title and content of the dialog correctly. NVDA on Chrome, however, does not. This is a pretty simplistic example of an alert dialog, where the readout should, in theory, work.

I would suggest opening an issue against NVDA to get the correct readout for an alert dialog. I hope that helps and let us know if you have any other questions!

**Comment by rulyu** [comment]
@natalieethell Thanks for your explanation. I have opened an issue against NVDA 
https://github.com/nvaccess/nvda/issues/10186

**Comment by natalieethell** [comment]
You're welcome! Thanks for opening an issue in NVDA. Let us know if you have any other questions about Dialog; otherwise, feel free to close this issue.
